wdl-analysis/src/types/v1.rs

```diff
@@ -531,6 +533,14 @@ pub struct ExprTypeEvaluator<'a, C> {
     /// If the count is non-zero, special evaluation behavior is enabled for
     /// string interpolation.
     placeholders: usize,
+
+    /// A cache of regexes that have been evaluated.
+    ///
+    /// This is used to avoid re-evaluating the same regex multiple times.
+    ///
+    /// The key is the regex string and the value is either an error or
+    /// a compied regex.
+    regex_cache: HashMap<String, Result<regex::Regex, String>>,
```

Comment:
The lifetime of `ExprTypeEvaluator` is typically pretty short so this cache won't live for very long.

I wonder if we could keep the cache in `populate_document` as a `HashMap<&str, Result<regex::Regex, regex::Error>>` and pass that on down in places where we construct an `ExprTypeEvaluator`.

I'm also wondering if it's truly worth having a cache as it only saves us on regular expressions that are repeatedly used throughout a single WDL document, and I'm unsure if it's worth the effort now, to be honest.

Reply:
Note that the actual WDL stdlib function implementations in the engine also don't currently do any sort of interning of the regular expressions passed to it, so I am leaning more towards just removing the cache in analysis until such time that it proves necessary.